Added new feature Schur_Number

slacker404 opened the pull request:
<!-- I have added a new feature in the combinatorics module the Schur_number -->


The Schur number S(k) is the largest integer n for which the interval  [1,n] can be partitioned into k sum-free sets.  http://mathworld.wolfram.com/SchurNumber.html

I have also made the partition which can be proven by induction and I have added test cases

slacker404 committed:
added feature Schur_number

slacker404 committed:
misc typos

asmeurer commented on sympy/combinatorics/tests/test_schur_number.py:
```diff
@@ -0,0 +1,31 @@
+        for j in subset:
+            if i + j in subset:
+                raise AssertionError("This subset is not sum free")
+    return "Everything is correct"
```
This line is not necessary

asmeurer commented on sympy/combinatorics/tests/test_schur_number.py:
```diff
@@ -0,0 +1,31 @@
+"""
+Checks if the occurance of all numbers  is exactly one
+"""
+t = 0
```
Tests should go inside of a test function. See how the other test files in SymPy are organized.

asmeurer commented:
Would it also be useful for this function to work symbolically?

slacker404 committed:
Added symbolic function for Schur_lower_bound and modified test_schur_number

slacker404 commented:
I have changed the files now and I added the symbolic representation of the inequality. I think that the result of the function cannot be represented symbolically because  there is no formula derived except from the last subset which has numbers of the form 3*n + 1

czgdp1807 commented:
It looks like a nice feature. @slacker404 Would you like to continue your work on this?

sachin-4099 commented:
I would like to work on this. Please guide as to what is remaining to be implemented.
@asmeurer

mijo2 commented:
I would like to revive this PR, and I am planning to do it today. @sachin-4099 We can work on this together if its ok with you, it is fine by me.

sachin-4099 commented:
> I would like to revive this PR, and I am planning to do it today. @sachin-4099 We can work on this together if its ok with you, it is fine by me.

It is fine by me..

czgdp1807 commented:
Closing in favour of https://github.com/sympy/sympy/pull/18667
